spec/filter_disk_spec.rb

```diff
@@ -6,52 +6,50 @@
 
 describe 'filter/disk' do
   before(:all) do
-    @tmp_dir = Dir.mktmpdir(nil, '/tmp')
-    FileUtils.chmod 'go+rwx', @tmp_dir
+    @tmp_dir = LKP::TmpDir.new('filter-disk-spec-rb-')
+    @tmp_dir.add_permission
 
-    @test_yaml_file = "#{@tmp_dir}/test.yaml".freeze
+    @test_yaml_file = "#{@tmp_dir.to_s}/test.yaml".freeze
   end
 
   after(:all) do
-    FileUtils.remove_entry @tmp_dir
+    @tmp_dir.cleanup!
+  end
+
+  def generate_job(contents = { 'testcase' => 'testcase' })
```

Comment:
looks no need to define default value of contents, it's ok to always require the caller to specify the contents.